core/domain/classifier_services.py

```diff
@@ -127,6 +127,99 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def create_classifier_training_jobs(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration domain object.
+        state_names: list(str). List of state names.
+    """
+    job_dicts_list = []
+    exp_id = exploration.id
+    exp_version = exploration.version
+    states = exploration.states
+    for state_name in state_names:
+        state = states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        job_dicts_list.append({
+            'algorithm_id': algorithm_id,
+            'interaction_id': interaction_id,
+            'exp_id': exp_id,
+            'exp_version': exp_version,
+            'state_name': state_name,
+            'training_data': training_data,
+            'status': feconf.TRAINING_JOB_STATUS_NEW
+        })
+
+    # Validate all the jobs.
+    for job_dict in job_dicts_list:
+        dummy_classifier_training_job = classifier_domain.ClassifierTrainingJob(
+            'job_id_dummy', job_dict['algorithm_id'],
+            job_dict['interaction_id'], job_dict['exp_id'],
+            job_dict['exp_version'], job_dict['state_name'],
+            job_dict['status'], job_dict['training_data'])
+        dummy_classifier_training_job.validate()
+
+    # Create all the classifier training jobs.
+    job_ids = classifier_models.ClassifierTrainingJobModel.create_multi(
+        job_dicts_list)
+
+    # Create mapping for each job.
+    mapping_dicts_list = []
+    for job_id_index, job_id in enumerate(job_ids):
+        mapping_dicts_list.append({
+            'exp_id': job_dicts_list[job_id_index]['exp_id'],
+            'exp_version': job_dicts_list[job_id_index]['exp_version'],
+            'state_name': job_dicts_list[job_id_index]['state_name'],
+            'job_id': job_id})
+    _create_multi_job_exploration_mappings(mapping_dicts_list)
+
+
+def create_job_exploration_mappings(exploration, state_names, change_list):
+    """Creates new TrainingJobExplorationMappingModel instances for all the
+    state names passed into the function. The mapping is created from the
+    new state to the ClassifierTrainingJob of the old state. If there's been
+    a change in the state name, we retrieve the old state name and create the
+    mapping accordingly.
+
+    Args:
+        exploration: Exploration. The Exploration domain object.
+        state_names: list(str). List of state names.
+        change_list: list(dict). A list of changes introduced in this commit.
+    """
+    mapping_dicts_list = []
+    exp_id = exploration.id
+    current_exp_version = exploration.version
+    old_exp_version = current_exp_version - 1
```

Comment:
@seanlip Yep. It can't be 1. This method- create_mappings() is never called when its version is 1. Thats because we only create new jobs for that version (and this handles mapping creation). I added a mention in the doctoring about that.